pkg/controller/topology.go

```diff
@@ -238,37 +268,56 @@ func GenerateAccessibilityRequirements(
 	requisiteTerms = slices.CompactFunc(requisiteTerms, slices.Equal)
 	// TODO (verult) reduce subset duplicate terms (advanced reduction)
 
+	// Filter out empty topology terms to prevent issues with caching and spreading.
+	// An empty term can be generated from an empty TopologySelectorTerm,
+	// but should not be considered for placement.
+	requisiteTerms = filterEmptyTerms(requisiteTerms)
 	requirement.Requisite = toCSITopology(requisiteTerms)
 
 	// 3. Generate CSI Preferred Terms
 	var preferredTerms []topologyTerm
-	if selectedCSINode == nil {
+	if len(selectedNodeName) == 0 {
 		// Immediate binding, we fallback to statefulset spreading hash for backwards compatibility.
+		preferredTerms, err = getPreferredTermsFromCache(pvcNodeStore, pvcUID)
```

Comment:
Do we actually need to cache preferred terms? Preferred is calculated from requisite, and shouldn't require looking up csinode/node info.

Reply:
Removed